setup.py

```diff
@@ -60,14 +60,16 @@
         'azdev.utilities',
     ],
     install_requires=[
+        'azure-cli-core',
+        'virtualenv',
```

Comment:
why we need this dependency?

Reply:
The virtualenv dependency? This is needed because the user can now use azdev setup to create a new virtual environment for them.

Reply:
From python 3, `python -m env venv` could do the same job. I think it's not necessary to introduce `virtualenv`

Reply:
This needed if for example in Windows you do pip install azdev and you do not have the virtaulenv module installed. The azdev setup would then fail if the user tried to run: azdev setup -c azure-cli -r azure-cli-extension -s tenv
because virtualenv package is not installed the -s param would fail when trying to make the virtual enviroment.

Reply:
Got it. Some installation of Python is way to raw such that the necessary files for creating the complete virtualenv is missing